webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/checks/endconfig/migration/MigrateDirectiveWithSelectInputBroken.scala

```diff
@@ -0,0 +1,118 @@
+ * Controlled by presence of flag file /opt/rudder/etc/force_ncf_technique_update
+ */
+class MigrateDirectiveWithSelectInputBroken(
+    techniqueReader:       EditorTechniqueReader,
+    roDirectiveRepository: RoDirectiveRepository,
+    writeDirectives:       WoDirectiveRepository,
+    uuidGenerator:         StringUuidGenerator
+) extends BootstrapChecks {
+
+  override val description = "Migrate directives with invalid parameters"
+
+  def updateDirectivesWithSelectInputBroken: ZIO[Any, RudderError, Unit] = {
+    val modificationId = ModificationId(uuidGenerator.newUuid)
+    for {
+      res        <- techniqueReader.readTechniquesMetadataFile
+      techniques  = res._1.filter(_.parameters.exists(_.constraints.exists(_.select.isDefined)))
+      directives <- ZIO.foreach(techniques) { t =>
+                      val selectParameters = t.parameters.filter(_.constraints.exists(_.select.isDefined))
+                      for {
+                        res                 <- roDirectiveRepository
+                                                 .getActiveTechnique(TechniqueName(t.id.value))
+                                                 .map(_.map(at => (at.id, at.directives)).getOrElse((ActiveTechniqueId("notFound"), Nil)))
+                        (atId, directiveIds) = res
+                        directives          <- ZIO.foreach(directiveIds)(roDirectiveRepository.getDirective).map(_.flatten)
+                        directiveToUpdate    = directives.filter(_.parameters.exists(p => selectParameters.exists(_.name == p._1)))
+                        updatedDirectives    = {
+                          directiveToUpdate.map(d => {
+                            selectParameters.foldLeft(d) {
+                              case (directive, param) =>
+                                directive.parameters.get(param.name) match {
+                                  case None        => directive
+                                  case Some(value) =>
+                                    directive.copy(parameters =
+                                      directive.parameters.removed(param.name).updated(param.id.value.toUpperCase, value)
+                                    )
+                                }
+                            }
+                          })
+                        }
+                        saved               <- ZIO.foreach(updatedDirectives)(d => {
+                                                 writeDirectives.saveDirective(
+                                                   atId,
+                                                   d,
+                                                   modificationId,
+                                                   RudderEventActor,
+                                                   Some(s"Updating invalid parameters in directive ${d.id.serialize}")
+                                                 )
+                                               })
+
```

Comment:
I would like to have a log that looks like a check (ie, don't tell user it migrates if there is nothing to migrate), and if migration did happen, then list UUIDs of impacted directive. 
Something lile: 

````
.... Checking if some directive have a select paramenter needing migration
... Migrated N directives: UUID1, UUID2, etc
````

Without that, it is quite difficult to know is any directives were migrated, so it can lead to harder than needed debug session. 
We can't rely on event log: I thought I could find them back by searching on the save message, but it does not work (which is a pitty).

Reply:
I hope this log line will be helpful to someone, I'm not sure a Logger for a migration for a case that may not really exists needed this much !! but it is done